Addidng example recipe and documentation for personal diagnostic (outside `diag_scripts`)

valeriupredoi opened the pull request:
This branch allows a not-so-advanced ESMValTool user to write their own Python module diagnostic and run it via an ESMValTool recipe that can call upon all the power of the Preprocessor, only that they don't need to commit anything to git beforehand or not even install ESMValTool. I need input from advanced users like @ledm or @schlunma to make the diagnostic wrapper from ```/esmvaltool/diag_scripts/personal_diagnostic.py``` as comprehensive as possible but not too comprehensive so that users will be lazy and not be willing to contribute as full-blown developers (right now, that wrapper returns just a preprocessed data dictionary). This is a bit of a sensitive topic in that we need such a capability for ESMValTool but we have to be careful users don't overuse it and the development of the tool stagnates.

schlunma commented:
Pretty interesting diagnostic @valeriupredoi! I think if we ensure that the structure of the external script is similar to a "real" diagnostic (one `main` function which accepts a `cfg`-like dictionary as only parameter) we should not have too much problems because porting is then fairly easy. Besides, for real applications the user needs to write a complete recipe anyway with all datasets and preprocessors, and I think in this case it is easier to use the standard approach with a regular diagnostic.

In addition, I think this could also be very handy to test little scripts with the preprocessed data of the ESMValTool :+1:

valeriupredoi commented:
@bouweandela cheers, dude, I wish I knew about this beforehand :open_mouth: -- anyways, can you import any of esmvaltool modules in your `test.py` so you can run a proper diagnostic-like script that will only be needed to be included in `diag_scripts` after you've decided to be a developer? If yes, than this PR has just become obsolete :cry:

bouweandela commented:
Hi V.,

Yes, just try it out, if esmvaltool is installed you can use it like any other installed Python module. Some of this PR may have become obsolete, but the documentation part is very valuable, we lack good beginner (or even developer, like yourself) documentation, so keep up the good work there!

valeriupredoi commented:
hey @bouweandela man, I tried running with your settings it ain't working - see the recipe:
```
datasets:
  - {dataset: MPI-ESM-LR,  project: CMIP5, exp: historical,  ensemble: r1i1p1,  start_year: 2000,  end_year: 2002}
  - {dataset: NorESM1-M,   project: CMIP5, exp: historical,  ensemble: r1i1p1,  start_year: 2001,  end_year: 2005}

preprocessors:
  pp:
    regrid:
      target_grid: 1x1
      scheme: linear

diagnostics:
  simple:
    description: "Simple personal diagnostic"
    variables:
      ta:
        preprocessor: pp
        field: T3M
        mip: Amon
        fx_files: [sftlf, sftof, areacello]
    scripts:
      my_diagnostic:
        script: /group_workspaces/jasmin2/cmip6_prep/esmvaltool_users/valeriu/my_little_diagnostic.py
```
it doesn't produce any work/ directory and the log file is empty, and no output either. Have you tested this setting, man?

valeriupredoi commented:
ow wait, I am doing smth wrong...ignore the previous post :grin:

valeriupredoi commented:
OK @bouweandela the show works like a charm with just specifying `script: /path/to/your/grandma/house.py` and I removed my initial wrapper and example module; added example recipe, diagnostic and readme markup file

valeriupredoi commented:
cheers @mattiarighi
